### DISCUSSION TITLE
README.md relative links base URL on npmjs

### DISCUSSION THREAD

**Discussion by aminnairi** [npm/feedback#210] [Website] [answered]
## Description

It is really convenient to write links in the `README.md` file as relative links.

For instance, in my `README.md`, I redirect people to the `LICENSE` file by using the relative link `./LICENSE` instead of the full URL like `https://github.com/user/repo/LICENSE`.

This is very well reflected on GitHub (obviously) and should also be the case in npmjs.

## Current behavior

The current behavior is that, for the links that are relatives, they all point to the npmjs website, instead of the GitHub one.

## Expected solution

Maybe there should be something like a detection in the `package.json` file for the `repository.url` property and use it to link to all relative URLs? And maybe some sort of clever fallback to detect where the repository is located (in case the property is missing), and just use the current behavior otherwise?

**Comment by ethomson** [comment, marked as answer]
We should respect the `repository.url` and `repository.directory` fields when creating links to content in GitHub repositories.  Can you point me to your package where this isn't working?

**Reply by aminnairi** [reply]
I guess the parser had not enough time to parse the `README.md` file when first publishing but it is now working as intended. Thanks for your time.

**Reply by ethomson** [reply]
Thanks for the update @aminnairi !
